### PR TITLE
PluginBrowser: Send also OS and architecture

### PR DESCRIPTION
**Pull request by thopiekar**
Orientated a bit on the way how browsers are sending their strings.
No idea, whether there is a standard on this or not.
Needs to be discussed I guess..

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Devs, see CURA-4246.

**Comment by thopiekar** [comment]
So easily merged? :scream: 
I thought you would be more interested in sending custom headers for each info. This would make it more easy to parse the info across different Cura versions.
However, no problem that I made the first step with this PR :stuck_out_tongue_winking_eye:

**Comment by alekseisasin** [comment]
It is first step, depends on it what will you do with this data. I wanted
extend Header data with "platform.version()" to know which windows version
is used. Also possible to add extra private data.

On 28 August 2017 at 18:11, Thomas Karl Pietrowski <notifications@github.com

**Comment by fieldOfView** [comment]
In any case, a common user agent would be nice. Uranium uses one, Cura now uses another (and OctoPrintPlugin uses yet another one).

**Comment by Ghostkeeper** [comment]
Should they all be using Cura's user agent?

**Comment by thopiekar** [comment]
Well, I wouldn't do that. I expect that parsing the user agent string will be needed at the end. So yeah, on Cura's side it is nice to put everything together, but on the server side I can imagine more work needs to be done.

However, I guess this needs to be discussed with people who are working on the web applications at Ultimaker.

**Comment by alekseisasin** [comment]
I don't think parsing will be an issue. Just choose delimiter like " ; "
and extend Header data.

File: plugins/PluginBrowser/PluginBrowser.py [hunk 1]
```diff
@@ -43,7 +44,14 @@ def __init__(self, parent = None):
 
         self._is_downloading = False
 
-        self._request_header = [b"User-Agent", str.encode("%s - %s" % (Application.getInstance().getApplicationName(), Application.getInstance().getVersion()))]
+        self._request_header = [b"User-Agent",
+                                str.encode("%s\%s (%s %s)" % (Application.getInstance().getApplicationName(),
```
Comment (LipuFei) [inline plugins/PluginBrowser/PluginBrowser.py]: This should be `<product>/<product-version>`. The splash here is not right... I will fix it.

Comment (LipuFei) [reply plugins/PluginBrowser/PluginBrowser.py]: Fixed in 0358b1fca

**Comment by LipuFei** [comment]
I think if we need extra data, that should be sent through the request body, not the HTTP headers. And the plugin server will be also a real (RESTful) service which processes requests based on the those data.

File: plugins/PluginBrowser/PluginBrowser.py (hunk 1 above)
Comment (thopiekar) [reply plugins/PluginBrowser/PluginBrowser.py]: Oh, thanks! 👍

**Pull request merged**